### PR TITLE
Add centralized Metasploit plugin documentation

### PR DESCRIPTION
**Pull request by adfoster-r7**
Add centralized Metasploit plugin documentation to the Metasploit Wiki

image

## Verification

- [x] Verify CI passes
- [x] Verify the docs are cromulent

### CODE REVIEW

File: docs/metasploit-framework.wiki/How-To-Use-Plugins.md [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+| token_adduser    | Attempt to add an account using all connected Meterpreter session tokens                            |
+| token_hunter     | Search all active Meterpreter sessions for specific tokens                                          |
+| wiki             | Outputs stored database values from the current workspace into dokuwiki or mediawiki format         |
+| wmap             | Web assessment plugin                                                                               |
+
+## Examples
+
+### Alias Plugin
+
+The Alias plugin adds the ability to alias console commands:
```
Comment (adfoster-r7) [inline docs/metasploit-framework.wiki/How-To-Use-Plugins.md:60]: We should look into a way of embedding references to separate markdown files if possible, obviously not erb - but jekyll's liquid equivalent:

```
<%- Dir['./plugins'].each do |file|
   <%= render file %>
<%- end %>
```

Or similar

File: plugins/session_notifier.rb [hunk 2]
```diff
@@ -400,7 +400,7 @@ def cleanup
     end
 
     def desc
-      'This plugin notifies you a new session via SMS.'
+      'This plugin notifies you a new session via SMS'
```
Comment (gwillcox-r7) [inline plugins/session_notifier.rb:403]:
```suggestion
      'This plugin notifies you of a new session via SMS'
```

File: plugins/wiki.rb [hunk 3]
```diff
@@ -476,7 +476,7 @@ def name
     # more than 60 characters, but there are no hard limits.
     #
     def desc
-      'Adds output to wikitext'
+      'Outputs stored database values from the current workspace into dokuwiki or mediawiki format'
```
Comment (gwillcox-r7) [inline plugins/wiki.rb:479]:
```suggestion
      'Outputs stored database values from the current workspace into DokuWiki or MediaWiki format'
```

File: docs/metasploit-framework.wiki/How-To-Use-Plugins.md [hunk 4]
```diff
@@ -0,0 +1,155 @@
+Metasploit plugins can change the behavior of Metasloit framework by adding new features, new user interface commands, and more.
```
Comment (gwillcox-r7) [inline docs/metasploit-framework.wiki/How-To-Use-Plugins.md:1]:
```suggestion
Metasploit plugins can change the behavior of Metasploit framework by adding new features, new user interface commands, and more.
```

File: docs/metasploit-framework.wiki/How-To-Use-Plugins.md [hunk 5]
```diff
@@ -0,0 +1,155 @@
...
+Plugins can be automatically loaded and configured on msfconsole's start up by configuring a custom `~/.msf4/msfconsole.rc` file:
+
+```
+load plugin_name
+plugin_name_command --option
+```
+
+## Available Plugins
+
+The current Available Framework can be found by running the `load -l` command, or viewing Metasploit's [plugins](https://github.com/rapid7/metasploit-framework/tree/master/plugins) directory:
```
Comment (gwillcox-r7) [inline docs/metasploit-framework.wiki/How-To-Use-Plugins.md:19]:
```suggestion
The current available plugins for Metasploit can be found by running the `load -l` command, or viewing Metasploit's [plugins](https://github.com/rapid7/metasploit-framework/tree/master/plugins) directory:
```

File: docs/metasploit-framework.wiki/How-To-Use-Plugins.md [hunk 6]
```diff
@@ -0,0 +1,155 @@
...
+| session_tagger   | Automatically interacts with new sessions to create a new remote TaggedByUser file                  |
+| socket_logger    | Log socket operations to a directory as individual files                                            |
+| sounds           | Automatically plays a sound when various framework events occur                                     |
+| sqlmap           | sqlmap plugin for Metasploit                                                                        |
+| thread           | Internal test tool for testing thread usage in Metasploit'                                          |
+| token_adduser    | Attempt to add an account using all connected Meterpreter session tokens                            |
+| token_hunter     | Search all active Meterpreter sessions for specific tokens                                          |
+| wiki             | Outputs stored database values from the current workspace into dokuwiki or mediawiki format         |
+| wmap             | Web assessment plugin                                                                               |
+
```
Comment (gwillcox-r7) [inline docs/metasploit-framework.wiki/How-To-Use-Plugins.md:55]:
```suggestion
| wmap             | Web assessment plugin                                                                               |
|------------------|-----------------------------------------------------------------------------------------------------|
```

File: docs/metasploit-framework.wiki/How-To-Use-Plugins.md [hunk 7]
```diff
@@ -0,0 +1,155 @@
...
+load alias
+alias proxy_enable "set Proxies http:localhost:8079"
+alias proxy_disable "unset Proxies"
+alias routes "route print"
+```
+
+### Capture Plugin
+
+Capturing credentials is a critical and early phase in the playbook of many offensive security testers. Metasploit has
+facilitated this for years with protocol-specific modules all under the auxiliary/server/capture. Users can start and configure
```
Comment (gwillcox-r7) [inline docs/metasploit-framework.wiki/How-To-Use-Plugins.md:114]:
```suggestion
facilitated this for years with protocol-specific modules all under the `modules/auxiliary/server/capture` directory. Users can start and configure
```

**Review by gwillcox-r7 — CHANGES_REQUESTED** [review]

File: docs/metasploit-framework.wiki/How-To-Use-Plugins.md (hunk 6 above)
Comment (adfoster-r7) [reply docs/metasploit-framework.wiki/How-To-Use-Plugins.md:55]: This isn't required for Github flavored markdown tables https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/organizing-information-with-tables - unless I'm missing something 👀

Comment (gwillcox-r7) [reply docs/metasploit-framework.wiki/How-To-Use-Plugins.md:55]: Yep your correct, my bad. Marking resolved!

**Comment by gwillcox-r7** [comment]
LGTM will land once tests pass

**Review by gwillcox-r7 — APPROVED** [review]

File: docs/metasploit-framework.wiki/How-To-Use-Plugins.md [hunk 8]
```diff
@@ -0,0 +1,155 @@
...
+| request          | Make requests from within Metasploit using various protocols.                                       |
+| rssfeed          | Create an RSS feed of events                                                                        |
+| sample           | Demonstrates using framework plugins                                                                |
+| session_notifier | This plugin notifies you of a new session via SMS                                                      |
+| session_tagger   | Automatically interacts with new sessions to create a new remote TaggedByUser file                  |
+| socket_logger    | Log socket operations to a directory as individual files                                            |
+| sounds           | Automatically plays a sound when various framework events occur                                     |
+| sqlmap           | sqlmap plugin for Metasploit                                                                        |
+| thread           | Internal test tool for testing thread usage in Metasploit'                                          |
```
Comment (gwillcox-r7) [inline docs/metasploit-framework.wiki/How-To-Use-Plugins.md:50]:
```suggestion
| thread           | Internal test tool for testing thread usage in Metasploit                                          |
```

File: docs/metasploit-framework.wiki/How-To-Use-Plugins.md [hunk 9]
```diff
@@ -0,0 +1,155 @@
...
+### Capture Plugin
+
+Capturing credentials is a critical and early phase in the playbook of many offensive security testers. Metasploit has
+facilitated this for years with protocol-specific modules all under the `modules/auxiliary/server/capture` directory. Users can start and configure
+each of these modules individually, but now the capture plugin can streamline the process. The capture plugin can easily start 13
+different services (17 including SSL enabled versions) on the same listening IP address including remote interfaces via Meterpreter.
+A configuration file can be used to select individual services to start and once finished, all services can easily be stopped
+using a single command.
+
+To use the plugin, it must first be loaded. That will provide the captureg command (for Capture-Global) which then offers start
```
Comment (gwillcox-r7) [inline docs/metasploit-framework.wiki/How-To-Use-Plugins.md:120]:
```suggestion
To use the plugin, it must first be loaded. That will provide the `captureg` command (for Capture-Global) which then offers start
```

**Review by gwillcox-r7 — CHANGES_REQUESTED** [review]

**Comment by gwillcox-r7** [comment]
Sorry quick rereview showed everything rendered properly but two minor issues came up, which I'll quickly fix now.

**Comment by gwillcox-r7** [comment]
# Release Notes
Additional documentation has been added to the Metasploit Wiki to explain how plugins work.

**Review by gwillcox-r7 — APPROVED** [review]

**Pull request merged**